core/src/main/java/org/kiwix/kiwixmobile/core/main/KiwixWebView.java

```diff
@@ -168,6 +168,7 @@ public void handleMessage(Message msg) {
       if (url != null || src != null) {
         url = url == null ? src : url;
         url = url.substring(url.lastIndexOf('/') + 1);
+        url = src == null || url.contains(".") ? url : src.substring(src.lastIndexOf('/') + 1);
```

Comment:
Maybe a new execution branch to make this clearer?
```
if(url!=null&& src!=null){
```

Reply:
That would mean a tonne of duplicate code?

Reply:
Yeah but we could clean this up a bunch and make whats happening clearer, the reassignment of the url variable is a bit irritating and we can just extract a method if the duplication becomes a concern.

We probably want a method that returns the url we actually want to parse `selectNonNullUrlWithExtension`? As of right now at first glance I have no idea what this code does, what src/url represent or why it fixes the problem and I have to mentally run through the execution branch to figure out what the url variable is.

It might be able to be expressed well in kotlin
```
pathSegment = 
url?.substringAfterLast("/")?.takeIf { it.contains(".") }
?: src?.substringAfterLast("/")?.takeIf { it.contains(".") }
 ?: url.?.substringAfterLast("/")
 ?: src?.substringAfterLast("/")
```
which I think is the flow here?